### PR TITLE
feat(options): add new options to avoid stale based on comments

### PR DESCRIPTION
**Pull request by C0ZEN**
## Changes

- [x] Add 3 new options to avoid staling when a comment is added
- [x] The scope of `remove-stale-when-updated` will be reduced, resulting in a breaking change (@luketomlinson for the release notes!) when using the option at `false`

⚠️ this is a breaking change

## Context

Helping to close (maybe) #441, #470, #435?
Closes #390 due to no activity

### CODE REVIEW

**Comment by C0ZEN** [comment]
I'm kind of stuck with the issue it caused.
May need help...

**Comment by jeremystretch** [comment]
@C0ZEN what's the issue?

**Comment by C0ZEN** [comment]
It makes sense to have this feature but the current implementation is actually helping the case where an issue is marked as stale then still in the same workflow process it for closing process. In such case the freshly staled issue update date is set as today and with the current implementation provided in this PR it will consider the issue as no longer stale if a comment was added. If the new option to install when commented is disabled there is no problem

File: src/classes/issues-processor.ts [hunk 1]
```diff
@@ -615,28 +615,63 @@ export class IssuesProcessor {
...
       issueLogger.info(
-        `Remove the stale label since the $$type has a comment and the workflow should remove the stale label when updated`
+        LoggerService.white('└── '),
+        `Skipping the process since the $$type is now un-stale`
       );
-      await this._removeStaleLabel(issue, staleLabel);
 
       // Are there labels to remove or add when an issue is no longer stale?
       await this._removeLabelsWhenUnstale(issue, labelsToRemoveWhenUnstale);
       await this._addLabelsWhenUnstale(issue, labelsToAddWhenUnstale);
```
Comment (flaxel) [inline src/classes/issues-processor.ts:667]: There is the error which triggered the failing test. The option `labelsToAddWhenUnstale` is empty and therefore no label will be added.

File: __tests__/main.spec.ts [hunk 2]
```diff
@@ -2352,3 +2425,46 @@ test('processing an issue stale since less than the daysBeforeStale without a st
...
+    async p => (p === 1 ? TestIssueList : []),
+    async (): Promise<IComment[]> => Promise.resolve([]),
+    async () => new Date().toDateString()
+  );
+
+  // process our fake issue list
+  await processor.processIssues(1);
+
+  expect(processor.staleIssues).toHaveLength(1);
+  expect(processor.addedLabelIssues).toHaveLength(1);
```
Comment (flaxel) [inline __tests__/main.spec.ts:2468]:
```suggestion

```

**Comment by C0ZEN** [comment]
The last added test is actually failing because just after staling, the workflow continue and then consider the issue as stale but to unstale due to `shouldRemoveStaleWhenUpdated && issueHasUpdate` being both `true`.

**Comment by C0ZEN** [comment]
I close it because I just don't understand the whole workflow.
Each time I change something a test is failing, and actually, I can't define if its bad tests or a bad changes.
The fact that the specs are mocking the label creation date and all the logic to calculate if either or not it should stale or close seems terrible IMO.
What could be nice is to rewrite this whole part — code and specs included — and I just don't have the courage to do so.
Well, free to continue the work from this branch or just tackle this from scratch.

**Pull request closed without merging**